politeiad/backendv2/tstorebe/store/localdb/encrypt.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package localdb
+
+import (
+	"bytes"
+
+	"github.com/decred/politeia/util"
+	"github.com/marcopeereboom/sbox"
+)
+
+func (l *localdb) encrypt(data []byte) ([]byte, error) {
+	l.keyMtx.RLock()
+	defer l.keyMtx.RUnlock()
+
+	return sbox.Encrypt(0, l.key, data)
+}
+
+func (l *localdb) decrypt(data []byte) ([]byte, uint32, error) {
+	l.keyMtx.RLock()
+	defer l.keyMtx.RUnlock()
+
+	return sbox.Decrypt(l.key, data)
+}
+
+func (l *localdb) zeroKey() {
+	l.keyMtx.Lock()
+	defer l.keyMtx.Unlock()
+
+	util.Zero(l.key[:])
+	l.key = nil
```

Comment:
check key == nil in other places

Reply:
Done

https://github.com/lukebp/politeia/blob/tlog/politeiad/backendv2/tstorebe/store/mysql/mysql.go#L62-L88

Reply:
This is not what I meant.

```
type localdb struct {
  key [32]byte // no pointer
}

// caller must zero key after setting it
func (l *localdb) SetKey(k *[32]byte) {
  copy(l.key[:], k[:])
}

// Now functions can use &l.key and we can toss the mutex.
```

Reply:
I fixed this.